fidimag/micro/zeeman.py

```diff
@@ -129,20 +130,136 @@ def time_fun(pos, t):
 
     """
 
-    def __init__(self, H0, time_fun, name='TimeZeeman'):
+    def __init__(self, time_fun, name='TimeZeeman'):
+        self.time_fun = time_fun
+        self.name = name
+        self.jac = True
 
-        self.H0 = H0
+    def setup(self, mesh, spin, Ms):
+        self.mesh = mesh
+        self.spin = spin
+        self.n = mesh.n
+
+        self.Ms = Ms
+        self.Ms_long = np.zeros(3 * mesh.n)
+
+        # TODO: Check if it is necessary to define a 3D matrix for
+        # the Ms vectors. Maybe there is a way that uses less memory
+        # (see the calculation in the *compute_energy* function)
+        self.Ms_long.shape = (3, -1)
+        for i in range(mesh.n):
+            self.Ms_long[:, i] = Ms[i]
+
+        self.Ms_long.shape = (-1,)
+        self.field = np.zeros(3 * self.n)
+
+    def compute_field(self, t=0, spin=None):
+        self.field[:] = helper.init_vector_func_fast(self.time_fun,
+                                                     self.mesh,
+                                                     False,
+                                                     t)
+        return self.field
+
+
+class SimpleTimeZeeman(Zeeman):
```

Comment:
Did you test this function?

Reply:
Refactored this, it shouldn't have been used here.